Tests/Tests/STPPaymentMethodTupleTest.m

```diff
@@ -0,0 +1,96 @@
+                                                                        availablePaymentTypes:@[[STPPaymentMethodType creditCard],
+                                                                                                [STPPaymentMethodType giropay]]
+                                                                        selectedPaymentMethod:source];
+
+    XCTAssertEqualObjects(tuple.selectedPaymentMethod, source);
+}
+
+- (void)testAllowedAvailableSelectedPaymentMethod {
+
+    STPCard *card = [STPCard decodedObjectFromAPIResponse:[STPTestUtils jsonNamed:@"Card"]];
+    STPSource *source = [STPSource decodedObjectFromAPIResponse:[STPTestUtils jsonNamed:@"CardSource"]];
+
+    STPPaymentMethodTuple *tuple = [[STPPaymentMethodTuple alloc] initWithSavedPaymentMethods:@[card, source]
+                                                                        availablePaymentTypes:@[[STPPaymentMethodType creditCard],
+                                                                                                [STPPaymentMethodType giropay]]
+                                                                        selectedPaymentMethod:[STPPaymentMethodType giropay]];
+
+    XCTAssertEqualObjects(tuple.selectedPaymentMethod, [STPPaymentMethodType giropay]);
+}
+
+- (void)testNotAllowedAvailableSelectedPaymentMethod {
+
+    STPCard *card = [STPCard decodedObjectFromAPIResponse:[STPTestUtils jsonNamed:@"Card"]];
+    STPSource *source = [STPSource decodedObjectFromAPIResponse:[STPTestUtils jsonNamed:@"CardSource"]];
+
+    STPPaymentMethodTuple *tuple = [[STPPaymentMethodTuple alloc] initWithSavedPaymentMethods:@[card, source]
+                                                                        availablePaymentTypes:@[[STPPaymentMethodType creditCard],
+                                                                                                [STPPaymentMethodType giropay]]
+                                                                        selectedPaymentMethod:[STPPaymentMethodType creditCard]];
+
+    XCTAssertNil(tuple.selectedPaymentMethod);
+}
+
+- (void)testMissingSelectedPaymentMethod {
+
+    STPCard *card = [STPCard decodedObjectFromAPIResponse:[STPTestUtils jsonNamed:@"Card"]];
+    STPSource *source = [STPSource decodedObjectFromAPIResponse:[STPTestUtils jsonNamed:@"CardSource"]];
+
+    STPPaymentMethodTuple *tuple = [[STPPaymentMethodTuple alloc] initWithSavedPaymentMethods:@[card, source]
+                                                                        availablePaymentTypes:@[[STPPaymentMethodType creditCard],
+                                                                                                [STPPaymentMethodType giropay]]
+                                                                        selectedPaymentMethod:[STPPaymentMethodType sofort]];
+
+    XCTAssertNil(tuple.selectedPaymentMethod);
+}
+
+- (void)testOnlyOneAvailableUnselectablePaymentMethod {
```

Comment:
The tuple does not actually do this check. I could move the logic in here? But as written that filtering of invalid saved cards is done elsewhere.